packages/core/src/client.ts

```diff
@@ -260,6 +257,35 @@ export class Client<
     return this.data
   }
 
+  fetchProvider({ bust, chainId }: { bust?: boolean; chainId?: number } = {}) {
+    let provider_ = this.providers.get(chainId ?? -1)
```

Comment:
We can't force a `chainId` to the in-memory map since a consumer may not pass a `chainId` (e.g. `provider: () => getDefaultProvider()`), so we fall back to `-1`.